### PR TITLE
gsctl reference: New page on "create kubeconfig", more improvements

### PR DESCRIPTION
**Pull request by marians**
Towards a more complete and more useful `gsctl` reference.

The PR also contains some clean-ups here and there.

Pinging a few more people, as this might contain valuable information for gsctl users and support.

### CODE REVIEW

File: content/reference/gsctl/create-kubeconfig.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+If things go as expected, you'll see a few lines of output like the following:
+
+```nohighlight
+Creating new key pair…
+New key pair created with ID 248e90bf4… and expiry of 720 hours
+Certificate and key files written to:
+/Users/myself/.config/gsctl/certs/w6wn8-ca.crt
+/Users/myself/.config/gsctl/certs/w6wn8-248e90bf45-client.crt
+/Users/myself/.config/gsctl/certs/w6wn8-248e90bf45-client.key
+Switched to kubectl context 'giantswarm-w6wn8'
```
Comment (xh3b4sd) [inline content/reference/gsctl/create-kubeconfig.md]: IMO this blob is rather hard to parse for humans. Below we use line breaks and indentations. We could improve this here as well. Thinking out loud here.

File: content/reference/gsctl/create-kubeconfig.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+- The next four lines show where certificates and the private key have been stored. Despite the name "key pair", there are actually three files downloaded.
+  - The first (name schema: `<cluster_id>-ca.crt`) is the certificate authority file and technically it's not part of your personal key pair.
+  - The other two are private key and client certificate.
+  - All three files are stored under `<config_dir>/certs` and permissions are set so that only you have permission to read those files.
+- Then it says `Switched to kubectl context giantswarm-<cluster_id>`. If this was the first key pair you created for this cluster, then this context has just been added to your kubectl configuration. And it is now set as the current context to work with.
+
+Then `gsctl` suggests that you check access to your cluster using the `kubectl cluster-info` command. If things work fine, you should see an output like this:
```
Comment (xh3b4sd) [inline content/reference/gsctl/create-kubeconfig.md]: Since you just ran into it. What about adding some "known issue" or disclaimer that the cluster creation takes some time. As of now following the docs and being quicker than the guest cluster k8s API being up, you would see an EOF or some other error. This might make people think it is broken. 

In the long run we have to add some status report mechanism. Lets make sure we also have an issue for this story.

File: content/reference/gsctl/global-options.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+```nohighlight
+$ awsgsctl list clusters
+...
+
+$ inhousegsctl list clusters
+...
+```
+
+If you use Windows, there are several equivalents to the Unix/Linux `alias` function. [This StackOverflow article](https://stackoverflow.com/questions/20530996/aliases-in-windows-command-prompt) explains many of them.
```
Comment (xh3b4sd) [inline content/reference/gsctl/global-options.md]: I wanted to make windows jokes but saw the repo is public. :D Keeping it professional for now.

File: content/reference/gsctl/create-cluster.md [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
 weight = 20
 +++
 
-# Creating a new Cluster using `gsctl`
+# `gsctl create cluster`
 
-With Giant Swarm, an organization can own any number of clusters. `gsctl create cluster` allows for creating a Kubernetes cluster using two different approaches:
+With Giant Swarm, an organization can own any number of clusters. `gsctl create cluster` allows for creating a new Kubernetes cluster.
+
+You can make usee of two different approaches:
```
Comment (oponder) [inline content/reference/gsctl/create-cluster.md]: Typo: `usee` -> `use`

File: content/reference/gsctl/create-kubeconfig.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    kubectl config use-context giantswarm-w6wn8
+```
+
+Let's go through the output piece by piece:
+
+- The first line is pretty obvious.
+- A `New key-pair created …` means that a new combination of a private key and a client certificate, in short "key pair", has been generated for you. Each key pair has a unique identifier, which is shown here in a truncated version. And a key pair doesn't live forever. The time to live is indicated in that row, too.
+- The next four lines show where certificates and the private key have been stored. Despite the name "key pair", there are actually three files downloaded.
+  - The first (name schema: `<cluster_id>-ca.crt`) is the certificate authority file and technically it's not part of your personal key pair.
+  - The other two are private key and client certificate.
```
Comment (oponder) [inline content/reference/gsctl/create-kubeconfig.md]: Typo: `The other two are private key and client certificate.` -> `The other two are the private key and client certificate.`

File: content/reference/gsctl/create-kubeconfig.md (hunk 2 above)
Comment (oponder) [inline content/reference/gsctl/create-kubeconfig.md]: typo: remove `an` from `an output like this`.

File: content/reference/gsctl/global-options.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
++++
+title = "Global command line options in gsctl"
+description = "Documentation for gsctl options that can be passed on the command line for all (or most) commands. Plus some tips on how you can use them to switch environments."
```
Comment (oponder) [inline content/reference/gsctl/global-options.md]: how about: `almost all commands`

File: content/reference/gsctl/create-kubeconfig.md (hunk 1 above)
Comment (marians) [inline content/reference/gsctl/create-kubeconfig.md]: Yeah, maybe we can make better use of empty lines in gsctl. Will try that some time, then adapt the docs.

File: content/reference/gsctl/create-kubeconfig.md (hunk 2 above)
Comment (marians) [inline content/reference/gsctl/create-kubeconfig.md]: I'll add a note on waiting for the cluster to come up.

File: content/reference/gsctl/global-options.md (hunk 3 above)
Comment (marians) [inline content/reference/gsctl/global-options.md]: Highly appreciated :)

**Comment by marians** [comment]
> Are these docs somehow auto generated or is this all manual? Can we maybe improve this a bit?

@xh3b4sd They are not, and I don't see a useful way to do that.

**Comment by marians** [comment]
Thanks for your reviews!
